source/common/event/file_event_impl.h

```diff
@@ -41,5 +45,119 @@ class FileEventImpl : public FileEvent, ImplBase {
   // polling and activating new fd events.
   const bool activate_fd_events_next_event_loop_;
 };
+
+// Forward declare for friend class.
+class UserSpaceFileEventFactory;
+
+// The interface of populating event watcher and obtaining the active events. The events includes
+// Read, Write and Closed.
+class EventListener {
+public:
+  virtual ~EventListener() = default;
+
+  // Provide the activated events.
+  virtual uint32_t triggeredEvents() PURE;
+  virtual uint32_t getAndClearEphemeralEvents() PURE;
+
+  // Callbacks of the event operation.
+  virtual void onEventEnabled(uint32_t enabled_events) PURE;
+  virtual void onEventActivated(uint32_t activated_events) PURE;
+};
+
+// Return the enabled events except EV_CLOSED. This implementation is generally good since only
+// epoll supports EV_CLOSED but the entire envoy code base supports another poller. The event owner
+// must assume EV_CLOSED is never activated. Also event owner must tolerate that OS could notify
+// events which are not actually triggered.
+class DefaultEventListener : public EventListener {
```

Comment:
why default?  isn't this more of a synthetic event handler?

Reply:
I think I replied but showed in inline thread.

The background is DefaultEventListener is returning many "might be triggered" events.
This DefaultEventListener has two roles here:
1. Correct though less efficient for this PR to merge.
2. When the advanced event listener is improving, DefaultEventListener acts as a backup, switchable by Envoy runtime.